rllib/algorithms/algorithm_config.py

```diff
@@ -901,17 +910,23 @@ def validate(self) -> None:
                 # compatibility for now. User only needs to set num_rollout_workers.
                 self.input_config["parallelism"] = self.num_rollout_workers or 1
 
-        # resolve rl_module_spec class
-        if self._enable_rl_module_api and self.rl_module_spec is None:
-            self.rl_module_spec = self.get_default_rl_module_spec()
-            if not isinstance(
-                self.rl_module_spec, (SingleAgentRLModuleSpec, MultiAgentRLModuleSpec)
-            ):
-                raise ValueError(
-                    "rl_module_spec must be an instance of "
-                    "SingleAgentRLModuleSpec or MultiAgentRLModuleSpec."
-                    f"Got {type(self.rl_module_spec)} instead."
-                )
+        if self._enable_rl_module_api:
+            if self.rl_module_spec is not None:
+                # Merge provided RL Module spec class with defaults
+                _check_rl_module_spec(self.rl_module_spec)
+                default_rl_module_spec = self.get_default_rl_module_spec()
+                # We can only merge if we have SingleAgentRLModuleSpecs.
+                # TODO(Artur): Support merging for MultiAgentRLModuleSpecs.
+                if (
+                    type(default_rl_module_spec) is SingleAgentRLModuleSpec
```

Comment:
better:
```
if isinstance(default, Single) and isinstance(spec, Single):
    ...
else:
    # TODO (Artur): add support blah blah blah
    raise ValueError
```

Reply:
Can't raise a value error here right now because one BC has a single agent default RLModule config but a multi agent one is configured. The state of the code before this PR simply didn't check that. I'm not aware how this should be handled but think we should do this in another PR since it's starting to blow up a little.

Reply:
I did the isinstance() thing though!

Reply:
oh I see, we should only raise an error if default is multi-agent but single-agent is provided. The reverse is actually allowed now that I think about it.

Reply:
So what should be the behavior incase multi agent is provided and single agent is default?

Reply:
Update all single agent specs in the MARL spec with the default spec?

Reply:
I'm simply not updating the config in that case in the latest commit. Feel free to let me know if there should be special handling in your opinion.